openedx_caliper_tracking/tasks.py

```diff
@@ -0,0 +1,90 @@
+"""
+Contains tasks related to Openedx Caliper Tracking.
+"""
+import json
+import logging
+import random
+
+from celery.task import task
+from django.conf import settings
+from kafka import KafkaProducer
+from kafka.errors import KafkaError
+
+from openedx.features.ucsd_features.utils import send_notification
+
+LOGGER = logging.getLogger(__name__)
+KAFKA_SETTINGS = settings.CALIPER_KAFKA_SETTINGS
+DEFAULT_FROM_EMAIL = settings.DEFAULT_FROM_EMAIL
+MAXIMUM_RETRIES = KAFKA_SETTINGS.get('MAXIMUM_RETRIES')
+ERROR_REPORT_EMAIL = KAFKA_SETTINGS.get('ERROR_REPORT_EMAIL')
+KAFKA_END_POINT = KAFKA_SETTINGS.get('END_POINT')
+KAFKA_TOPIC_NAME = KAFKA_SETTINGS.get('TOPIC_NAME')
+
+
+@task(bind=True, max_retries=MAXIMUM_RETRIES)
+def deliver_caliper_event_to_kafka(self, transformed_event, event_type):
+    """
+    Deliver caliper event to kafka.
+
+    Retries for the given number of max_tries in case of any error else
+    sends an error report to the specified email address.
+    """
+    try:
+        LOGGER.info('Attempt # {} of sending event: {} to kafka is in progress.'.format(
+                    self.request_stack().get('retries'), event_type))
```

Comment:
@tasawernawaz This line is responsible for logging on each attempt for sending logs to kafka.